### PR TITLE
133 enhance contributing guidelines

### PR DESCRIPTION
**Pull request by Sandared**
This is a PR for issue #133 , enhancing the contributing guidelines.

Work done:
* restructured CONTRIBUTING.md
* added a table of contents
* added explaining text to *Reporting Issues* section
* added a template for issues
* added setup isntructions for Git, Ruby, Bundler and Jekyll
* added a troubleshooting section for Jekyll on Windows
* added some minor explanations to *Contribution guidelines* section

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,43 +1,122 @@
...
+**We will appreciate it.**
 
-Run jekyll in the root directory to build:
+The site uses [GitHub Pages](https://help.github.com/articles/what-are-github-pages/) with the site content in the `gh-pages` branch. To build the site locally, you will need to install Ruby, Bundler and Jekyll. If you've never worked with GitHub Pages before, just keep reading.
 
-	$ bundle exec jekyll build
+Or you already are a pro at working with GitHub Pages? You have set up the [git triangular workflow](https://www.sociomantic.com/blog/2014/05/git-triangular-workflow/) for this repository and have Ruby, Bundler and Jekyll running? Then what are you waiting for? Read the [contributing guidelines](#contributing-guidelines) section and start writing!
 
-You can also run a local server to test the site:
+Github pages can be created either with usual HTML or you can use [Jekyll](https://jekyllrb.com/), a static web site generator, which takes [Markdown](https://jekyllrb.com/docs/posts/) files and transforms them into a static webpage. Usually you want to write in markdown, as this is way more convenient than writing pure HTML.
```
Comment (bjhargrave) [inline CONTRIBUTING.md]: I don't think the link for Markdown is good link to explain what markdown is. https://daringfireball.net/projects/markdown/ is a better one.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,43 +1,122 @@
...
-Then go to [http://localhost:4000](http://localhost:4000). The pages are automatically updated when you edit a markdown file, though you do have to refresh the browser to see these changes. Eclipse later revisions have a decent markdown editor build in.
+### Install Git, Ruby and Bundler
 
-We use [Travis CI](https://travis-ci.org/osgi/osgi.enroute.site) to test build any pull requests.
+**Update/Install Git**
+* Check if you've already installed Git: type `git --version` in a console
+* If a version is printed out it's fine (at the time of writing:  2.14.1)
+* If not:
+ 	* **Windows:** go to [Git](https://git-scm.com/) and download/install the current version
+	* **Linux:** type `apt-get install git-all` in a console
```
Comment (bjhargrave) [inline CONTRIBUTING.md]: You neglect macOS which may devs use. On macOS, you get git with XCode command line tools, or you can install via brew with `brew install git`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,43 +1,122 @@
...
 
-## Workflow
+**Update/Install Ruby**
+* Check if you've arleady installed Ruby: type `ruby --version` in a console
+* If a version is printed out it's fine (at the time of writing: 2.4.1)
+* If not:
+	* **Windows:** download [Ruby](https://rubyinstaller.org/) and install it
+	* **Linux:** type `apt-get install ruby` in a console
+> *There seems to be a problem with json 1.8.3 and Ruby 2.4.x. So you should either install 2.3.3 for now or follow these [instructions](https://github.com/github/pages-gem/issues/376)*
```
Comment (bjhargrave) [inline CONTRIBUTING.md]: Also macOS. XCode includes ruby 2.0 and I don't think we require a newer version than that. You can install ruby via brew also.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -47,7 +126,7 @@ This will clone the main repo to a local repo on your disk and set up the `origi
 Next you will set up the the second side of the triangle to your fork repo.
 
     cd osgi.enroute.site
-    git remote add fork git@github.com:github-user/osgi.enroute.site.git
+    git remote add fork https://github.com/<YourUserName>/osgi.enroute.site.git
```
Comment (bjhargrave) [inline CONTRIBUTING.md]: Since you push to your clone/fork, you should use the ssh URL. I don't think the https URL is to be used for pushing.

Comment (Sandared) [inline CONTRIBUTING.md]: Is there any advantage in doing it over ssh? 

I've read the GitHub [help page](https://help.github.com/articles/which-remote-url-should-i-use/) on this topic and to me it seems ssh comes with quite an overhead compared to https. According to [GitHub Help](https://help.github.com/articles/connecting-to-github-with-ssh/) using ssh requires you to 
* generate the ssh key
* add it to the ssh agent
* add the ssh key to your GitHub account

whereas https works out of the box.

Both methods seem to require a password each time you do a `pull`, `push` or similar to a remote repository, unless your are using a credential helper.

With respect to inexperienced users/contributors (including me ;) ) I would favor a setup as simple as possible.

Comment (bjhargrave) [inline CONTRIBUTING.md]: Well GitHub does seem to recommend https: https://help.github.com/articles/which-remote-url-should-i-use/

But if you have 2FA for your your github credentials (and you do, dont you?), then you will need create a personal access token to use instead if your actual github password. 

At this point, ssh is a lot better. :-)

Comment (Sandared) [reply CONTRIBUTING.md]: Well until now I didn't had 2FA ;)
I added a section about setting up SSH and changed the url back to SSH :)

**Comment by bjhargrave** [comment]
Looks good. Can you please squash your commits into a single commit?

**Comment by Sandared** [comment]
I hope everything is correct now :)

**Comment by bjhargrave** [comment]
Thanks for the work!

**Pull request merged**